src/ir/module-splitting.cpp

```diff
@@ -355,9 +355,11 @@ void ModuleSplitter::setupJSPI() {
   // Support the first version of JSPI, where the JSPI pass added the load
   // secondary module export.
   // TODO: remove this when the new JSPI API is only supported.
-  if (primary.getExportOrNull(LOAD_SECONDARY_MODULE)) {
+  if (primary.getExportOrNull(LOAD_SECONDARY_MODULE) &&
+      primary.getExport(LOAD_SECONDARY_MODULE)->kind ==
+        ExternalKind::Function) {
```

Comment:
We can reduce the number of lookups here:
```suggestion
  if (auto* loadSecondary = primary.getExportOrNull(LOAD_SECONDARY_MODULE);
      loadSecondary && loadSecondary->kind == ExternalKind::Function) {
```